share/completions/pip2.fish

```diff
@@ -1,5 +1,2 @@
-# pip[2|3] emits (or emitted) completions with the wrong command name
-# See discussion at https://github.com/fish-shell/fish-shell/pull/4448
-# and pip bug at https://github.com/pypa/pip/pull/4755
-# Keep this even after pip fix is upstreamed for users not on the latest pip
-pip2 completion --fish 2>/dev/null | string replace -r -- " -c\s+pip\b" " -c pip2" | source
+__fish_cache_sourced_completions pip2 completion --fish 2>/dev/null
+or pip2 completion --fish 2>/dev/null | source
```

Comment:
nit: we could probably wrap it in another function that does the
`or pip2 completion --fish 2>/dev/null | source` automagically

Reply:
I don't want to add a second function file for that, and adding it to the main function is awkward because you would have to add it everywhere we return, or wrap it in an if.

Reply:
ok that's all fine. Maybe if we had anonymous functions.

```fish
set -l lambda $(function --anonymous -V foo; ...; end)
$lamdba || error
```